code-execution/src/main/java/com/dojo/codeexecution/controller/GameResultController.java

```diff
@@ -20,15 +21,19 @@
     private final RestTemplate restTemplate;
     private final CodenjoyConfigProperties codenjoyConfigProperties;
 
+    private final DockerClient dockerClient;
+
     @Autowired
-    public GameResultController(DockerEventUpdateHandler dockerEventUpdateHandler, RestTemplate restTemplate, CodenjoyConfigProperties codenjoyConfigProperties) {
+    public GameResultController(DockerEventUpdateHandler dockerEventUpdateHandler, RestTemplate restTemplate, CodenjoyConfigProperties codenjoyConfigProperties, DockerClient dockerClient) {
         this.dockerEventUpdateHandler = dockerEventUpdateHandler;
         this.restTemplate = restTemplate;
         this.codenjoyConfigProperties = codenjoyConfigProperties;
+        this.dockerClient = dockerClient;
     }
 
     @PostMapping(path = "/test/result")
     public void testResult(@RequestBody TestResult testResult) {
+                dockerClient.stopContainerCmd(testResult.getContainerId()).exec();
```

Comment:
check if not already stopped and check after calling stopContainer the stop was successful